:farmer: `test_node__rmw` test failing for `connextdds`, `cyclonedds` and `fastrtps`

## Bug report

**Required Info:**

- Operating System: RHEL 9.2
- Installation type: Source
- Version or commit hash: Rolling
- DDS implementation: All

#### Steps to reproduce issue

1. Run a build in https://ci.ros2.org/view/nightly/job/nightly_linux-rhel_debug/
2. See tests fail

#### Expected behavior

All tests should pass

#### Actual behavior

Test node tests are failing for each rmw implementation:
* [rcl.TestNodeFixture.test_rcl_node_init_with_internal_errors](https://ci.ros2.org/view/nightly/job/nightly_linux-rhel_debug/1734/testReport/junit/rcl/TestNodeFixture/test_rcl_node_init_with_internal_errors/)
* [projectroot.test.test_node__rmw_connextdds](https://ci.ros2.org/view/nightly/job/nightly_linux-rhel_debug/1734/testReport/junit/projectroot/test/test_node__rmw_connextdds/)
* [projectroot.test.test_node__rmw_cyclonedds_cpp](https://ci.ros2.org/view/nightly/job/nightly_linux-rhel_debug/1734/testReport/junit/projectroot/test/test_node__rmw_cyclonedds_cpp/)
* [projectroot.test.test_node__rmw_fastrtps_cpp](https://ci.ros2.org/view/nightly/job/nightly_linux-rhel_debug/1734/testReport/junit/projectroot/test/test_node__rmw_fastrtps_cpp/)
* [rcl.test_node__rmw_connextdds.gtest.missing_result](https://ci.ros2.org/view/nightly/job/nightly_linux-rhel_debug/1734/testReport/junit/(root)/rcl/test_node__rmw_connextdds_gtest_missing_result/)
* [rcl.test_node__rmw_cyclonedds_cpp.gtest.missing_result](https://ci.ros2.org/view/nightly/job/nightly_linux-rhel_debug/1734/testReport/junit/(root)/rcl/test_node__rmw_cyclonedds_cpp_gtest_missing_result/)

#### Additional information

Reference build: https://ci.ros2.org/view/nightly/job/nightly_linux-rhel_debug/1734/

Log output:
```
The following tests FAILED:
        16 - test_node__rmw_connextdds (Failed)
        45 - test_node__rmw_cyclonedds_cpp (Failed)
        74 - test_node__rmw_fastrtps_cpp (Failed)
```

<details>
<summary>
Output for test_node__rmw_connextdds, test_node__rmw_cyclonedds_cpp and test_node__rmw_fastrtps_cpp
</summary>

```
16: [WARNING] mocking_utils::inject_on_return() cannot forward call to original 'rmw_destroy_node' function before injection^M
16:     at /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/test/rcl/test_node.cpp:522^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/snprintf.c:42 Injecting fault and executing {(*__errno_location ()) = 22; return -1;}^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/allocator.c:34 Injecting fault and returning ((void *)0)^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/snprintf.c:42 Injecting fault and executing {(*__errno_location ()) = 22; return -1;}^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/allocator.c:59 Injecting fault and returning ((void *)0)^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/node_options.c:48 Injecting fault and executing { do {rcutils_set_error_state("Injecting " "11", "/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/node_options.c", 48);} while (0); return 11; }^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c:301 Injecting fault and executing { do {rcutils_set_error_state("Injecting " "11", "/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c", 301);} while (0); return 11; }^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c:302 Injecting fault and executing { do {rcutils_set_error_state("Injecting " "201", "/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c", 302);} while (0); return 201; }^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c:303 Injecting fault and executing { do {rcutils_set_error_state("Injecting " "10", "/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c", 303);} while (0); return 10; }^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c:304 Injecting fault and executing { do {rcutils_set_error_state("Injecting " "1", "/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c", 304);} while (0); return 1; }^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c:321 Injecting fault and executing { do {rcutils_set_error_state("Injecting " "11", "/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c", 321);} while (0); return 11; }^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c:322 Injecting fault and executing { do {rcutils_set_error_state("Injecting " "202", "/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c", 322);} while (0); return 202; }^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c:323 Injecting fault and executing { do {rcutils_set_error_state("Injecting " "10", "/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c", 323);} while (0); return 10; }^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c:324 Injecting fault and executing { do {rcutils_set_error_state("Injecting " "1", "/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/remap.c", 324);} while (0); return 1; }^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/snprintf.c:42 Injecting fault and executing {(*__errno_location ()) = 22; return -1;}^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/allocator.c:34 Injecting fault and returning ((void *)0)^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/snprintf.c:42 Injecting fault and executing {(*__errno_location ()) = 22; return -1;}^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/allocator.c:34 Injecting fault and returning ((void *)0)^M
16: /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/snprintf.c:42 Injecting fault and executing {(*__errno_location ()) = 22; return -1;}^M
16: /home/jenkins-agent/work[ERROR] [1701251807.265286260] [rmw_connextdds]: failed to allocate RMW node^M
16: [ERROR] [1701251807.265367441] [rmw_connextdds]: failed to allocate node name^M
16: [ERROR] [1701251807.265383135] [rmw_connextdds]: failed to allocate node namespace^M
16: ^M
16: >>> [rcutils|error_handling.c:108] rcutils_set_error_state()^M
16: This error state is being overwritten:^M
16: ^M
16:   'failed to allocate memory for hash map impl, at /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/hash_map.c:279'^M
16: ^M
16: with this new error message:^M
16: ^M
16:   'Failed to initialize type cache hash map, at /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/node_type_cache.c:62'^M
16: ^M
16: rcutils_reset_error() should be called after error handling to avoid this.^M
16: <<<^M
16: ^M
16: >>> [rcutils|error_handling.c:108] rcutils_set_error_state()^M
16: This error state is being overwritten:^M
16: ^M
16:   'failed to allocate memory for map data, at /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcutils/src/hash_map.c:295'^M
16: ^M
16: with this new error message:^M
16: ^M
16:   'Failed to initialize type cache hash map, at /home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/src/ros2/rcl/rcl/src/rcl/node_type_cache.c:62'^M
16: ^M
16: rcutils_reset_error() should be called after error handling to avoid this.^M
16: <<<^M
16: [ERROR] [1701251807.266459283] [rmw_connextdds]: not all nodes finalized: 1^M
16: [ERROR] [1701251807.563092724] [rmw_connextdds]: not all nodes finalized: 1^M
16: terminate called without an active exception^M
16: -- run_test.py: return code -6^M
16: -- run_test.py: generate result file '/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/build/rcl/test_results/rcl/test_node__rmw_connextdds.gtest.xml' with failed test^M
16: -- run_test.py: verify result file '/home/jenkins-agent/workspace/nightly_linux-rhel_debug/ws/build/rcl/test_results/rcl/test_node__rmw_connextdds.gtest.xml'^M
1/3 Test #16: test_node__rmw_connextdds ........***Failed    2.13 sec^M

```

</details>

---

This is a consistent issue since [rhel_debug/1720](https://ci.ros2.org/view/nightly/job/nightly_linux-rhel_debug/1720)

> **Note:**
>
> This was after the pyside2 broken bindings, so the builds that have this failure are the same that use the pinned docker image

Ros2 repos diff
```diff
   eProsima/Fast-DDS:
     type: git
     url: https://github.com/eProsima/Fast-DDS.git (version 2.11.x)
-    version: 4ab9cc534fa4769d74836b086c1b7573c8e5736b
+    version: 12bb5b3662229490f8bc6c934c766efb92124023

   ros2/rcl:
     type: git
     url: https://github.com/ros2/rcl.git
-    version: bf5d4bcd85240068626630c78865e3e28ccce1b0
+    version: cc0cd52c398f953fbaba11b1a90b3416ce22b58a
```

--- a/rcl/test/CMakeLists.txt
+++ b/rcl/test/CMakeLists.txt
@@ -234,7 +234,7 @@ function(test_target)
   if(
     "${DISTRIBUTION}" STREQUAL "\"centos\"" OR
     "${DISTRIBUTION}" STREQUAL "\"almalinux\"")
-    set(gtest_filter_env_var "GTEST_FILTER=-TestNodeFixture__*.test_rcl_node_init_with_internal_errors")
+    set(gtest_filter_env_var "GTEST_FILTER=-TestNodeFixture.test_rcl_node_init_with_internal_errors")
   else()
     set(gtest_filter_env_var "")
   endif()

